examples/decoding/plot_decoding_time_generalization.py

```diff
@@ -12,80 +12,81 @@
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+events_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+raw = mne.io.Raw(raw_fname, preload=True)
+picks = mne.pick_types(raw.info, meg=True, exclude='bads')  # Pick MEG channels
+raw.filter(1, 30, method='fft')  # Band pass filtering signals
+events = mne.read_events(events_fname)
+event_id = {'AudL': 1, 'AudR': 2, 'VisL': 3, 'VisR': 4}
+decim = 2  # decimate to make the example faster to run
+epochs = mne.Epochs(raw, events, event_id, -0.050, 0.400, proj=True,
+                    picks=picks, baseline=None, preload=True,
+                    reject=dict(mag=5e-12), decim=decim, verbose=False)
 
-raw = mne.io.Raw(raw_fname % 1, preload=True)  # Take first run
-raw.append(mne.io.Raw(raw_fname % 2, preload=True))  # Take second run too
+###############################################################################
+# Generalization across time (GAT)
 
-picks = mne.pick_types(raw.info, meg=True, exclude='bads')
-raw.filter(1, 45, method='iir')
+# The function implements the method used in:
+# King, Gramfort, Schurger, Naccache & Dehaene, "Two distinct dynamic modes
+# subtend the detection of unexpected sounds", PLOS ONE, 2013
 
-events = mne.find_events(raw, stim_channel='UPPT001')
-event_id = {"faces": 1, "scrambled": 2}
-tmin, tmax = -0.1, 0.5
+# Define events of interest
+events = epochs.events[:, 2]
+viz_vs_auditory = np.in1d(events, (1, 2)).astype(int)
 
-# Set up pick list
-picks = mne.pick_types(raw.info, meg=True, eeg=False, stim=True, eog=True,
-                       ref_meg=False, exclude='bads')
+gat = GeneralizationAcrossTime(predict_mode='cross-validation')
 
-# Read epochs
-decim = 4  # decimate to make the example faster to run
-epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
-                    picks=picks, baseline=None, preload=True,
-                    reject=dict(mag=1.5e-12), decim=decim)
+# fit and score
+gat.fit(epochs, y=viz_vs_auditory)
+gat.score(epochs, y=viz_vs_auditory)
+gat.plot_diagonal()  # plot decoding across time (correspond to GAT diagonal)
+gat.plot()  # plot full GAT matrix
 
-epochs_list = [epochs[k] for k in event_id]
-mne.epochs.equalize_epoch_counts(epochs_list)
 
 ###############################################################################
-# Run decoding
+# Generalization across time and across conditions
 
-# Compute Area Under the Curver (AUC) Receiver Operator Curve (ROC) score
-# of time generalization. A perfect decoding would lead to AUCs of 1.
-# Chance level is at 0.5.
-# The default classifier is a linear SVM (C=1) after feature scaling.
-scores = time_generalization(epochs_list, clf=None, cv=5, scoring="roc_auc",
-                             shuffle=True, n_jobs=2)
+# As proposed in King & Dehaene (2014) 'Characterizing the dynamics of mental
+# representations: the temporal generalization method', Trends In Cognitive
+# Sciences, 18(4), 203-210.
```

Comment:
I am tempted to split the example in 2 as I feel this new example much more advanced than the old one and we should fight complexity. That way we would be able to still use SPM data which is nicer to demo the method. Example would then be simple short and nice.

wdyt?